gapic/samplegen/samplegen.py

```diff
@@ -280,14 +280,17 @@ def validate_and_transform_request(self,
                     # and requires less munging of the assigned value
                     duplicate["value"] = f"'{val}'"
                     break
+                elif attr.is_primitive:
+                    # Only valid if this is the last attribute in the chain.
+                    break
                 else:
                     raise TypeError
 
             if i != len(attr_chain) - 1:
-                # We broke out of the loop after processing an enum.
+                # We broke out of the loop after processing an enum or a primitive.
                 extra_attrs = ".".join(attr_chain[i:])
                 raise types.InvalidEnumVariant(
```

Comment:
Let's include something about primitives in this error name?

Reply:
Good catch!